attools/src/SwiftBuildTool.swift

```diff
@@ -13,71 +13,195 @@
-        s += "        print(\"extension \\(self.dynamicType) : XCTestCaseProvider {\")\n"
-        s += "        print(\"    var allTests : [(String, () -> Void)] {\")\n"
-        s += "        print(\"        return [\")\n"
-        s += "        print(\"        (\\\"testFoo\\\", testFoo)\")\n"
-        s += "        print(\"        ]\")\n"
-        s += "        print(\"    }\")\n"
-        s += "        print(\"}\")\n"
-        s += "        print(\"(Or disable xctestStrict.)\")\n"
-        s += "        print(\"Cheers! -- Anarchy Tools Team\")\n"
-        s += "    }\n"
-        s += "    override public func tearDown() {\n"
-        s += "        if let provider = self as? XCTestCaseProvider {\n"
-        s += "            let contains = provider.allTests.contains({ test in\n"
-        s += "                return test.0 == invocation!.selector.description\n"
-        s += "            })\n"
-        s += "            if !contains {\n"
-        s += "               XCTFail(\"Test \\(name) is missing from \\(self.dynamicType)\")\n"
-        s += "               implementAllTests()\n"
-        s += "            }\n"
-        s += "        }\n"
-        s += "        else {\n"
-        s += "            XCTFail(\"Whoops!  \\(self.dynamicType) doesn't conform to XCTestCaseProvider.\")\n"
-        s += "            implementAllTests()\n"
-        s += "        }\n"
-        s += "        super.tearDown()\n"
-        s += "    }\n"
-        s += "}\n"
-        s += "\n"
-        return s
-    }()
+/**
+ * The `llbuild.swift` file contains all of the tools necessary to work
+ * with the swift-build-tool (llbuild) low-level build system. There are
+ * three tools contained here:
+ *
+ *    1. llbuild-config - this tool is reponsible for generate the build
+ *       configuration file necessary for llbuild to consume.
+ *    2. llbuild-build - this tool is responsible for consuming a build
+ *       configuration file and producing the appropriate outputs.
+ *    3. llbuild - this is a wrapper tool that provides a simple syntax
+ *       to merge the previous two tools together.
+ */
+
+/**
+ * This tool handles the generation of the configuration file for the
+ * llbuild tool (`swift-build-tool`).
+ */
+final class SwiftBuildToolConfig: Tool {
+    enum Keys {
+        static let Tool = "tool"
+        static let Name = "name"
+        static let ModuleName = "module-name"
+        static let Depedencies = "dependencies"
+        static let OutputType = "output-type"
+        static let Sources = "sources"
+        static let ConfigFile = "config-file"
+        static let CompileOptions = "compile-options"
+        static let LinkOptions = "link-options"
+        static let LinkSDK = "link-sdk"
+        static let LinkWithProduct = "link-with"
+        static let SwiftPath = "swiftc-path"
+        static let XCTestSupport = "xctestify"
+        static let XCTestStrict = "xctest-strict"
+    }
```

Comment:
We need to update the documentation at https://github.com/AnarchyTools/atbuild/blob/master/docs/atllbuild.md